File: java/Film.java
// ----------------------------------------------------
// CLASS FILM
// Blueprint objek data film dengan atribut private (Enkapsulasi)
// ----------------------------------------------------
public class Film {
    // Atribut privat (menyembunyikan informasi internal)
    private String id;
    private String judul;
    private String genre;
    private int durasi;

    // Konstruktor kosong
    public Film() {}

    // Konstruktor berparameter untuk instansiasi objek dengan nilai awal
    public Film(String id, String judul, String genre, int durasi) {
        this.id = id;
        this.judul = judul;
        this.genre = genre;
        this.durasi = durasi;
    }

    // Getter dan Setter untuk akses atribut private
    public String getId() { 
        return id; 
    }
    public void setId(String id) { 
        this.id = id; 
    }

    public String getJudul() { 
        return judul; 
    }
    public void setJudul(String judul) { 
        this.judul = judul; 
    }

    public String getGenre() { 
        return genre; 
    }
    public void setGenre(String genre) { 
        this.genre = genre; 
    }

    public int getDurasi() { 
        return durasi; 
    }
    public void setDurasi(int durasi) { 
        this.durasi = durasi; 
    }

    // Method perilaku untuk menampilkan informasi objek
    public void tampilkanInfo() {
        System.out.println("ID: " + id + " | Judul: " + judul + " | Genre: " + genre + " | Durasi: " + durasi + " menit");
    }
}

File: cpp/Film.cpp
#include <iostream>
#include <string>

using namespace std;

// ----------------------------------------------------
// CLASS FILM
// Merepresentasikan entitas data film dengan enkapsulasi
// ----------------------------------------------------
class Film {
private:
    // Atribut privat (melindungi data dari akses langsung luar class)[cite: 1]
    string id;
    string judul;
    string genre;
    int durasi;

public:
    // Konstruktor kosong
    Film() {}

    // Konstruktor berparameter untuk inisialisasi awal objek[cite: 1]
    Film(string id, string judul, string genre, int durasi) {
        this->id = id;
        this->judul = judul;
        this->genre = genre;
        this->durasi = durasi;
    }

    // Getter dan Setter untuk mengakses dan mengubah atribut privat[cite: 1]
    string getId() { 
        return id; 
    }
    void setId(string id) { 
        this->id = id; 
    }

    string getJudul() { 
        return judul; 
    }
    void setJudul(string judul) { 
        this->judul = judul; 
    }

    string getGenre() { 
        return genre; 
    }
    void setGenre(string genre) { 
        this->genre = genre; 
    }

    int getDurasi() { 
        return durasi; 
    }
    void setDurasi(int durasi) { 
        this->durasi = durasi; 
    }

    // Method untuk menampilkan informasi lengkap film[cite: 1]
    void tampilkanInfo() {
        cout << "ID: " << id << " | Judul: " << judul << " | Genre: " << genre << " | Durasi: " << durasi << " menit\n";
    }

    // Destruktor[cite: 1]
    ~Film() {}
};
